### PR TITLE
feature/AKD-285: Plugin to check bucket pay per requester enabled and related unit tests

### PR DESCRIPTION
**Pull request by UmairAjmal**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -795,6 +795,7 @@ module.exports = {
         'openVNCServer'                 : require(__dirname + '/plugins/alibaba/ecs/openVNCServer.js'),
       
         'bucketLoggingEnabled'          : require(__dirname + '/plugins/alibaba/oss/bucketLoggingEnabled.js'),
+        'bucketRequestPayment'          : require(__dirname + '/plugins/alibaba/oss/bucketRequestPayment.js'),
```
Comment (AkhtarAmir) [inline exports.js:798]: This name should be same as plugin title

Comment (AkhtarAmir) [reply exports.js:798]:
```suggestion
        'bucketPayByRequester'          : require(__dirname + '/plugins/alibaba/oss/bucketPayByRequester.js'),
```

File: plugins/alibaba/oss/bucketRequestPayment.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        if (!listBuckets.data.length) {
+            helpers.addResult(results, 0, 'No OSS buckets found', region);
+            return callback(null, results, source);
+        }
+
+        async.each(listBuckets.data, (bucket, cb) => {
+            if (!bucket.name) return cb();
+            
+
```
Comment (AkhtarAmir) [inline plugins/alibaba/oss/bucketRequestPayment.js:37]:
```suggestion

```

File: plugins/alibaba/oss/bucketRequestPayment.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        if (!listBuckets.data.length) {
+            helpers.addResult(results, 0, 'No OSS buckets found', region);
+            return callback(null, results, source);
+        }
+
+        async.each(listBuckets.data, (bucket, cb) => {
+            if (!bucket.name) return cb();
+
```
Comment (AkhtarAmir) [inline plugins/alibaba/oss/bucketRequestPayment.js]:
```suggestion
```

File: plugins/alibaba/oss/bucketRequestPayment.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        if (!listBuckets.data.length) {
+            helpers.addResult(results, 0, 'No OSS buckets found', region);
+            return callback(null, results, source);
+        }
+
+        async.each(listBuckets.data, (bucket, cb) => {
+            if (!bucket.name) return cb();
+            
+
+            var getBucketRequestPayment = helpers.addSource(cache, source,
```
Comment (AkhtarAmir) [inline plugins/alibaba/oss/bucketRequestPayment.js]:
```suggestion
            if (!bucket.name) return cb();

            var getBucketRequestPayment = helpers.addSource(cache, source,
```

File: plugins/alibaba/oss/bucketRequestPayment.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            if (!getBucketRequestPayment || getBucketRequestPayment.err || !getBucketRequestPayment.data) {
+                helpers.addResult(results, 3,
+                    `Unable to query OSS bucket info: ${helpers.addError(getBucketRequestPayment)}`, bucketLocation, resource);
+                return cb();
+            }
+
+            if (getBucketRequestPayment.data.payer &&
+                getBucketRequestPayment.data.payer.toLowerCase() === 'requester') {
+                helpers.addResult(results, 0,
+                    'Bucket has pay by requester mode enabled',
```
Comment (AkhtarAmir) [inline plugins/alibaba/oss/bucketRequestPayment.js:52]:
```suggestion
                    'Bucket has pay-by-requester feature enabled',
```

File: plugins/alibaba/oss/bucketRequestPayment.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                    bucketLocation, resource);
+            } else {
+                helpers.addResult(results, 2,
+                    'Bucket does not have pay by requester mode enabled',
```
Comment (AkhtarAmir) [inline plugins/alibaba/oss/bucketRequestPayment.js:56]:
```suggestion
                    'Bucket does not have pay-by-requester feature enabled',
```

**Review by giorod3 — APPROVED** [review]

**Pull request merged**